apps/scan/backend/src/printing/printer.ts

```diff
@@ -0,0 +1,104 @@
+import { PrinterStatus as LegacyPrinterStatus } from '@votingworks/types';
+import { detectPrinter, Printer as LegacyPrinter } from '@votingworks/printing';
+import {
+  PrinterStatus as FujitsuPrinterStatus,
+  PrinterState as FujitsuPrinterState,
+  PrintResult as FujitsuPrintResult,
+  getFujitsuThermalPrinter,
+  FujitsuThermalPrinterInterface,
+} from '@votingworks/fujitsu-thermal-printer';
+import {
+  BooleanEnvironmentVariableName,
+  getEnvironmentVariable,
+} from '@votingworks/utils';
+import { BaseLogger } from '@votingworks/logging';
+import { assert, Result } from '@votingworks/basics';
+import { Buffer } from 'buffer';
+
+export const USE_FUJITSU_PRINTER = getEnvironmentVariable(
+  BooleanEnvironmentVariableName.SCAN_USE_FUJITSU_PRINTER
+);
+
+export type PrinterStatus =
+  | ({
+      scheme: 'hardware-v3';
+    } & LegacyPrinterStatus)
+  | ({
+      scheme: 'hardware-v4';
+    } & FujitsuPrinterStatus);
+
+export type { FujitsuPrinterState, FujitsuPrinterStatus, FujitsuPrintResult };
+
+/**
+ * An abstraction that wraps old CUPS-based printing and V4 hardware custom driver printing.
+ */
+export type Printer = {
+  getStatus: () => Promise<PrinterStatus>;
+} & (
+  | {
+      scheme: 'hardware-v3';
+      print(data: Uint8Array): Promise<void>;
+    }
+  | {
+      scheme: 'hardware-v4';
+      print(data: Uint8Array): Promise<Result<void, FujitsuPrinterStatus>>;
+    }
+);
+
+export type PrintResult =
+  | {
+      scheme: 'hardware-v3';
+      pageCount: number;
+    }
+  | {
+      scheme: 'hardware-v4';
+      result: FujitsuPrintResult;
+    };
```

Comment:
To be able to have a singular `Printer` in the app, and a single `getPrinterStatus` endpoint and `printReport` endpoint, creating all these discriminated types.